src/sentry/workflow_engine/migrations/0012_migrate_metric_alerts.py

```diff
@@ -0,0 +1,187 @@
+        # should I skip migrating snapshotted and disabled rules?
+        # what about the date_added and date_updated fields? can I write to those to preserve history?
+        # description column will be added to the Detector model, but at the time of writing it isn't there yet. will need to update this
+        # same for threshold_period, will likely be added to the Detector model but isn't there yet
+        # AlertRuleActivity creation rows need to be backfilled into the Detector model, needs new created_by and created_at columns
+
+        query_subscription = QuerySubscription.objects.get(snuba_query=rule.snuba_query_id)
+        data_source = DataSource.update_or_create(
+            organization_id=rule.organization_id,
+            query_id=query_subscription.id,
+            type=DataSourceType.SNUBA_QUERY_SUBSCRIPTION,
+        )
+        data_condition_group = DataConditionGroup.update_or_create(
+            logic_type=DataConditionGroupType.ANY,
+            organization_id=rule.organization_id,
+        )
+        workflow = Workflow.objects.update_or_create(
+            name=rule.name,
+            organization_id=rule.organization_id,
+            when_condition_group=data_condition_group.id,
+        )
+        detector = Detector.update_or_create(
+            organization_id=rule.organization_id,
+            name=rule.name,
+            data_sources=data_source,
+            workflow_condition_group=data_condition_group.id,
+            type="MetricDetector",  # shouldn't this be in an enum?
+            owner_user_id=rule.user_id,
+            owner_team=rule.team,
+        )
+
+        # state is based on incident status
+        project = AlertRuleProjects.objects.get(alert_rule_id=rule.id)
+        incident = Incident.objects.get_active_incident(
+            alert_rule=rule,
+            project=project,
+            subscription=query_subscription,
+        )
+        state = DetectorPriorityLevel.OK
+        if incident:
+            # we have 45 incidents with a status of OPEN
+            # but it's all for the same disabled rule, seems like an edge case
+            if incident.status == IncidentStatus.CRITICAL.value:
+                state = DetectorPriorityLevel.HIGH
+            elif incident.status == IncidentStatus.WARNING.value:
+                state = DetectorPriorityLevel.MEDIUM
+
+        DetectorState.update_or_create(
+            detector=detector.id,
+            active=False,  # is this based on if there's a RuleSnooze for all?
+            state=state,
+        )
+        triggers = AlertRuleTrigger.objects.filter(alert_rule_id=rule.id)
+        for trigger in triggers:
+            DataCondition.update_or_create(
+                condition=trigger.threshold_type,
+                comparison=trigger.alert_threshold,
+                condition_result=state,  # ??? is that right?
+                type="idk",  # talk to josh about this
```

Comment:
in issue alerts this is https://github.com/getsentry/sentry/blob/ceorourke/metrics-timer-seer-response/src/sentry/constants.py#L264, I'm not sure if we need more than 1 for metric alerts